addon_updater.py

```diff
@@ -46,1669 +45,1702 @@
+
+    # -------------------------------------------------------------------------
+    # Getters and setters
+    # -------------------------------------------------------------------------
+    @property
+    def addon(self):
+        return self._addon
+
+    @addon.setter
+    def addon(self, value):
+        self._addon = str(value)
+
+    @property
+    def api_url(self):
+        return self._engine.api_url
+
+    @api_url.setter
+    def api_url(self, value):
+        if not self.check_is_url(value):
+            raise ValueError("Not a valid URL: " + value)
+        self._engine.api_url = value
+
+    @property
+    def async_checking(self):
+        return self._async_checking
+
+    @property
+    def auto_reload_post_update(self):
+        return self._auto_reload_post_update
+
+    @auto_reload_post_update.setter
+    def auto_reload_post_update(self, value):
+        try:
+            self._auto_reload_post_update = bool(value)
+        except:
+            raise ValueError("auto_reload_post_update must be a boolean value")
+
+    @property
+    def backup_current(self):
+        return self._backup_current
+
+    @backup_current.setter
+    def backup_current(self, value):
+        if value is None:
+            self._backup_current = False
+            return
+        else:
+            self._backup_current = value
+
+    @property
+    def backup_ignore_patterns(self):
+        return self._backup_ignore_patterns
+
+    @backup_ignore_patterns.setter
+    def backup_ignore_patterns(self, value):
+        if value is None:
+            self._backup_ignore_patterns = None
+            return
+        elif not isinstance(value, list):
+            raise ValueError("Backup pattern must be in list format")
+        else:
+            self._backup_ignore_patterns = value
+
+    @property
+    def check_interval(self):
+        return (self._check_interval_enable,
+                self._check_interval_months,
+                self._check_interval_days,
+                self._check_interval_hours,
+                self._check_interval_minutes)
+
+    @property
+    def current_version(self):
+        return self._current_version
+
+    @current_version.setter
+    def current_version(self, tuple_values):
+        if tuple_values is None:
+            self._current_version = None
+            return
+        elif type(tuple_values) is not tuple:
+            try:
+                tuple(tuple_values)
+            except:
+                raise ValueError(
+                    "Not a tuple! current_version must be a tuple of integers")
+        for i in tuple_values:
+            if type(i) is not int:
+                raise ValueError(
+                    "Not an integer! current_version must be a tuple of integers")
+        self._current_version = tuple(tuple_values)
+
+    @property
+    def engine(self):
+        return self._engine.name
+
+    @engine.setter
+    def engine(self, value):
+        if value.lower() is "github":
```

Comment:
With local testing, using `is` breaks things here. Using the python interactive console in Blender 2.90 (python 3.7.7) this is the behavior:

`>>> value = "GitHub"
>>> value.lower() is "github"
False
>>> value.lower() == "github"
True``
```

Changing this back to `==` works locally for me.

Reply:
Merged this back in to my repo.